lib/puppet/defaults.rb

```diff
@@ -1570,7 +1570,10 @@ def self.default_diffargs
     :pluginsync => {
       :default    => true,
       :type       => :boolean,
-      :desc       => "Whether plugins should be synced with the central server.",
+      :desc       => "Whether plugins should be synced with the central server. This setting is
```

Comment:
Sorry, for not making PUP-5708 more clear (the desired behavior changed since that ticket was written).

For 4.x, we will preserve the current pluginsync behavior if the user specifies an explicit value. So if they say `pluginsync=false` then we won't pluginsync, even if we're using a cached catalog.

But if the user doesn't then the default value will be determined based on the `use_cached_catalog` setting.

So for this PR we just want to issue a deprecation warning if the user explicitly sets the property. I would remove the part about "this setting will be ignored and pluginsync will not be synced."

In 5.0, we will remove the `pluginsync` setting, and the agent will always pluginsync when getting a new catalog, or not pluginsync when applying a cached catalog.